.github/workflows/commit-check.yaml

```
@@ -0,0 +1,93 @@
+jobs:
+  validate-commit-messages:
+    name: Check Commit Messages
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Extract Branch Name
+        id: branch
+        run: echo "branch_name=${{ github.head_ref }}" >> $GITHUB_ENV
```

actionlint (1.7.4)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)